cirq/contrib/acquaintance/executor.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+from typing import Dict, TYPE_CHECKING, Sequence
+
+from cirq import abc
+from cirq.circuits import (
+        Circuit, PointOptimizer, PointOptimizationSummary)
+from cirq.ops import Operation, GateOperation, QubitId, OP_TREE
+from cirq.contrib.acquaintance.gates import AcquaintanceOpportunityGate
+from cirq.contrib.acquaintance.permutation import (
+        PermutationGate,
+        LogicalIndex,
+        LogicalGates, LogicalMapping)
+from cirq.contrib.acquaintance.strategy import (
+        AcquaintanceStrategy, expose_acquaintance_gates)
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from typing import Callable, List, DefaultDict
+
+class ExecutionStrategy(metaclass=abc.ABCMeta):
+    """An execution strategy."""
```

Comment:
This comment is just repeating the name. What is an execution strategy used for? What problem does it solve? Why would someone want to use it?